src/main/java/com/librato/disco/Decoder.java

```diff
@@ -0,0 +1,14 @@
+package com.librato.disco;
+
+/**
+ * Interface to decode bytes stored in Zookeeper to type T
+ * @param <T> type of object excepted from decoding bytes
+ */
+public interface Decoder<T> {
+    /**
+     * Decode a byte array into an object of type T
+     * @param bytes the byte array payload stored in Zookeeper
+     * @return returns the object as decoded from bytes
+     */
+    T decode(byte[] bytes);
```

Comment:
Could we have this return some superset of Optional that also provides information on the exception?